test/unit/setting_registry_branding_test.rb

```diff
@@ -27,14 +28,29 @@ class SettingRegistryBrandingTest < ActiveSupport::TestCase
     end
 
     test 'hides upstream-only settings' do
-      assert_nil Setting['allow_multiple_content_views']
+      Foreman::SettingManager.define('test2') do
+        category(:Content) do
+          setting(
+            :test_setting,
+            description: N_("Upstream-only setting test"),
+            default: 'definitely not falsey',
+            full_name: N_('Upstream-only setting test'),
+            type: :string
+          )
+        end
+      end
+      Foreman.settings.load
+      UpstreamOnlySettings.expects(:include?).with('test_setting').returns(true)
+      assert_nil Setting['test_setting']
     end
   end
+  # rubocop:enable Metrics/BlockLength
 end
 
 class SettingBrandingTest < ActiveSupport::TestCase
   test 'replaces warning for upstream-only settings' do
-    Rails.logger.expects(:debug).with('Setting \'allow_multiple_content_views\' is not available in Satellite; ignoring')
-    Setting['allow_multiple_content_views']
+    UpstreamOnlySettings.expects(:include?).with('test_setting').returns(true)
+    Rails.logger.expects(:debug).with('Setting \'test_setting\' is not available in Satellite; ignoring')
+    Setting['test_setting']
```

Comment:
You are not defining the `test_setting` in settings here. Are you relying on the test execution order?

Also not related to this specific PR: why do we have two classes in a single file? Can we separate them please?

Reply:
I think this relies on the fact that `Setting[something]` always queries the Settings registry and triggers the warning code, even if the setting is not actually defined. (The test worked also before the setting definition in the previous test that I asked for)